spanner_orm/tests/migrations_test.py

```diff
@@ -0,0 +1,121 @@
+# python3
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import os
+import unittest
+
+from spanner_orm import error
+from spanner_orm.admin import migration_manager
+
+
+class TestMigration(object):
+
+  def __init__(self, migration_id, prev_migration):
+    self._id = migration_id
+    self._prev = prev_migration
+
+  @property
+  def migration_id(self):
+    return self._id
+
+  @property
+  def prev_migration(self):
+    return self._prev
+
+  def upgrade(self):
+    pass
+
+  def downgrade(self):
+    pass
+
+
+class MigrationsTest(unittest.TestCase):
+  TEST_DIR = os.path.dirname(__file__)
+  TEST_MIGRATIONS_DIR = os.path.join(TEST_DIR, 'migrations')
+
+  def test_retrieve(self):
+    manager = migration_manager.MigrationManager(self.TEST_MIGRATIONS_DIR)
+    migrations = manager.migrations
+    self.assertEqual(len(migrations), 3)
+    self.assertEqual(migrations[2].prev_migration, migrations[1].migration_id)
+    self.assertEqual(migrations[1].prev_migration, migrations[0].migration_id)
+
+  def test_generate(self):
+    manager = migration_manager.MigrationManager(self.TEST_MIGRATIONS_DIR)
+    try:
+      path = manager.generate('test migration')
+      migration = manager._migration_from_file(path)
+      self.assertIsNotNone(migration.migration_id)
+      self.assertIsNotNone(migration.prev_migration)
+      self.assertIsNotNone(migration.upgrade)
+      self.assertIsNotNone(migration.downgrade)
+    except Exception as ex:
+      raise ex
+    finally:
+      os.remove(path)
+
+  def test_order_migrations(self):
+    first = TestMigration('1', None)
+    second = TestMigration('2', '1')
+    third = TestMigration('3', '2')
+    migrations = [third, first, second]
+    expected_order = [first, second, third]
+
+    manager = migration_manager.MigrationManager(self.TEST_MIGRATIONS_DIR)
+    self.assertEqual(manager._order_migrations(migrations), expected_order)
+
+  def test_order_migrations_with_no_none(self):
```

Comment:
this is so that people can delete old migrations and it will still work. in the migration side of the code, i'm going to make sure that the first migration's prev_migration_id is None or has already been migrated, so this should be fine.